File: Meduim/128. Longest Consecutive Sequence/C++/Solution.cpp
#include <iostream>
#include <vector>
#include <set>
using namespace std;

int longestConsecutive(vector<int>& nums) {
    set<int> mySet(nums.begin(), nums.end());
    int maximum = 0;
    for(const auto& elem : mySet){
        if (!mySet.count(elem - 1)){
            int actual = 1;
            while (mySet.count(elem + actual)){
                actual++;
            }
            if (actual > maximum){
                maximum = actual;
            }
        }
    }
    return maximum;
}

int main() {
    vector<int> nums1;
    nums1 = {100,4,200,1,3,2};
    cout << "The results of Test Case 1 " << longestConsecutive(nums1) << endl;

    vector<int> nums2;
    nums2 = {0,3,7,2,5,8,4,6,0,1};
    cout << "The results of Test Case 2 " << longestConsecutive(nums2) << endl;

    vector<int> nums3;
    nums3 = {1,0,1,2};
    cout << "The results of Test Case 3 " << longestConsecutive(nums3) << endl;
}	

File: Meduim/128. Longest Consecutive Sequence/Scala/Solution.scala
object Main {
    def longestConsecutive(nums: Array[Int]): Int = {
        var maximum = 0
        val mySet = nums.toSet

        for (elem <- mySet){
            // Try to find the beginning of the element
            if(!mySet.contains(elem-1)){
                var actual = 1

                while(mySet.contains(elem+actual)){
                    actual += 1
                }

                if (maximum < actual){
                    maximum = actual
                }
            
            }

        }

        return maximum
    }

    def main(args: Array[String]): Unit = {
        val nums1 = Array(100, 4, 200, 1, 3, 2)
        val nums2 = Array(0, 3, 7, 2, 5, 8, 4, 6, 0, 1)
        val nums3 = Array(1, 0, 1, 2)

        println(longestConsecutive(nums1))
        println(longestConsecutive(nums2))
        println(longestConsecutive(nums3))
    }
}